components/org.wso2.carbon.identity.mgt.store.connector.jdbc/src/main/java/org/wso2/carbon/identity/mgt/store/connector/jdbc/queries/MySQLFamilySQLQueryFactory.java

```diff
@@ -306,4 +310,42 @@ public String getQuerryForUserIdFromMultipleAttributes(List<Attribute> attribute
 
         return getUniqueUserQuerry.toString();
     }
+
+    public PreparedStatement getPreparedStatementFromMultipleAttributes(
+            List<Attribute> attributes, int offset, int length, UnitOfWork unitOfWork) throws SQLException {
+        StringBuilder getUniqueUserQuerry = new StringBuilder();
+        getUniqueUserQuerry.append("SELECT UM_USER.USER_UNIQUE_ID FROM UM_USER WHERE UM_USER.ID IN");
+        int count = 1;
+        while (count <= attributes.size()) {
```

Comment:
What will happen if the list of attributes is empty, i.e. size == 0 ?

Reply:
This is handled in the method which calls this method, Only if attributes.size()>0, the prepared statement is generated else throw new IdentityStoreConnectorException which tells that There are no attributes to find users. 
getUsers(List<Attribute> attributes, int offset, int length) --> JDBCIdentityStoreConnector.java

Reply:
This is a public method. There is no control over who calls this method.
Any method should give valid result on any range of input parameters or throw a checked exception if parameters are our of range. We can not assume the other party will call with correct parameters.